### PR TITLE
Application Provisioning Connection Profile update: add OAUTH2 support for authScheme

### PR DESCRIPTION
**Pull request by jennaxue-okta**
## Description:
- **What's changed?** Add OAuth 2.0 based authentication support for Provisioning Connection.
- **Is this PR related to a Monolith release?**  Yes, 2022.02.2

### Resolves:

* OKTA-456535

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 1]
```diff
@@ -8422,15 +8422,15 @@ The provisioning connection object is a read only object that displays the metho
...
 | _links            | Discoverable resources related to the connection            | [JSON HAL](http://tools.ietf.org/html/draft-kelly-json-hal-06)              | TRUE    | FALSE   | TRUE    |           |
 | status            | Status of the connection      | `ENABLED`, `DISABLED`, `UNKNOWN`  | FALSE    | FALSE   | TRUE    | `DISABLED` |
 
 If the authScheme is `UNKNOWN`, then either the authentication scheme used by the application isn't supported or the the application doesn't support provisioning. An object with an `UNKNOWN` `authScheme` results in an `UNKNOWN` `status`.
 
 ### Provisioning Connection Profile object
 
-The application provisioning connection profile is used to configure the method of authentication and the credentials. Currently, only token based authentication is supported.
+The application provisioning connection profile is used to configure the method of authentication and the credentials. Currently, both token based and  OAuth 2.0 based authentication <ApiLifecycle access="ea" /> are supported.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:8433]:
```suggestion
The application provisioning connection profile is used to configure the method of authentication and the credentials. Currently, both token-based and OAuth 2.0-based authentication <ApiLifecycle access="ea" /> are supported.
```

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 2]
```diff
@@ -8422,15 +8422,15 @@ The provisioning connection object is a read only object that displays the metho
...
 
 #### Token based Provisioning Connection Profile example
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:8435]:
```suggestion
#### Token-based Provisioning Connection Profile example
```

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 3]
```diff
@@ -8449,6 +8449,23 @@ The application provisioning connection profile is used to configure the method
 | authScheme              | Defines the method of authentication     | `TOKEN` | FALSE    | FALSE  | FALSE    |
 | token            | Token used to authenticate with application      | String | FALSE    | FALSE   | FALSE    |
 
+#### OAuth 2.0 based Provisioning Connection Profile example
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:8452]:
```suggestion
#### OAuth 2.0-based Provisioning Connection Profile example
```

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 4]
```diff
@@ -8449,6 +8449,23 @@ The application provisioning connection profile is used to configure the method
...
+```json
+{
+    "profile": {
+        "authScheme": "OAUTH2",
+        "clientId": "${clientId}"
+    }
+}
+```
+
+#### OAuth 2.0 based Provisioning Connection Profile properties
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:8462]:
```suggestion
#### OAuth 2.0-based Provisioning Connection Profile properties
```
Maybe add `<ApiLifecycle access="ea" />` below this header to apply EA flag to entire table?

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 5]
```diff
@@ -8449,6 +8449,23 @@ The application provisioning connection profile is used to configure the method
...
+
+| Property                                | Description                                                   | DataType                               | Nullable | Unique | Readonly |
+| --------------------------------------- | ------------------------------------------------------------- | -------------------------------------- | -------- | ------ | -------- | --------- | --------- | ---------- |
+| authScheme                              | Defines the method of authentication                          | `OAUTH2` <ApiLifecycle access="ea" />  | FALSE    | FALSE  | FALSE    |
+| clientId <ApiLifecycle access="ea" />   | Unique identifier for OAuth 2.0 service app in the target org | String                                 | FALSE    | FALSE  | FALSE    |
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:8467]:
```suggestion
| clientId <ApiLifecycle access="ea" />   | Unique client identifier for the OAuth 2.0 service app from the target org | String                                 | FALSE    | FALSE  | FALSE    |
```

**Review by vanngo-okta — APPROVED** [review]
LGTM! I'm approving with some minor suggestions.

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 6]
```diff
@@ -8422,17 +8422,17 @@ The provisioning connection object is a read only object that displays the metho
...
 | _links            | Discoverable resources related to the connection            | [JSON HAL](http://tools.ietf.org/html/draft-kelly-json-hal-06)              | TRUE    | FALSE   | TRUE    |           |
 | status            | Status of the connection      | `ENABLED`, `DISABLED`, `UNKNOWN`  | FALSE    | FALSE   | TRUE    | `DISABLED` |
 
 If the authScheme is `UNKNOWN`, then either the authentication scheme used by the application isn't supported or the the application doesn't support provisioning. An object with an `UNKNOWN` `authScheme` results in an `UNKNOWN` `status`.
 
 ### Provisioning Connection Profile object
 
-The application provisioning connection profile is used to configure the method of authentication and the credentials. Currently, only token based authentication is supported.
+The application provisioning connection profile is used to configure the method of authentication and the credentials. Currently, both token-based and OAuth 2.0-based authentication <ApiLifecycle access="ea" /> are supported.
```
Comment (timhowie-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:8433]: You might change:
"Currently, both token-based and OAuth...."
to
"Currently, token-based and OAuth..."

Just thinking we don't need to specify 'both' as we list which options are currently available in the description.

**Review by timhowie-okta — COMMENTED** [review]
lgtm; I made a small suggestion, but you can feel free to implement or ignore, as either way is ok IMO.

**Pull request merged**